### PR TITLE
[Conformance Test] Fix cache test case failure for auto plugin

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - check if the blob size remains the same as it was during the initial caching of the compiled model, rather than comparing it with a specified number, such as 1 in this case.
 - count the size of cached blobs after the model compilation is completed on all HW plugin within AUTO plugin.

### Tickets:
 - CVS-130395

### CODE REVIEW

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp [hunk 1]
```diff
@@ -242,22 +242,38 @@ void CompileModelCacheTestBase::run() {
...
         // Step 2: Load with cache. Export or import shall not throw
         compiledModel = {}; // Destroy network object
         inferRequest = {};
         {
             core->set_property(ov::cache_dir(m_cacheFolderName));
             ASSERT_NO_THROW(compiledModel = core->compile_model(function, targetDevice, configuration));
-            if (targetDevice.find("AUTO") == std::string::npos)
+            if (targetDevice.find("AUTO") == std::string::npos) {
                 // Apply check only for HW plugins
                 ASSERT_EQ(i != 0, compiledModel.get_property(ov::loaded_from_cache));
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:256]: btw, AUTO can support ov::loaded_from_cache as well, why skip here

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp [hunk 2]
```diff
@@ -242,22 +242,38 @@ void CompileModelCacheTestBase::run() {
...
+            } else {
+                while (true) {
+                    auto exeDevices = compiledModel.get_property(ov::execution_devices);
+                    if (exeDevices.size() == 1 && exeDevices.front().find("(CPU)") != std::string::npos)
+                        continue;
+                    break;
+                }
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:263]: the logic here seems not useful for the check below, no matter hardware is finished loading, you are not checking the exact blob number below

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp [hunk 3]
```diff
@@ -242,22 +242,38 @@ void CompileModelCacheTestBase::run() {
         GTEST_FAIL() << "Can't compile network without cache for " << m_functionName << " with precision " << m_precision.get_type_name() << std::endl;
     }
     auto originalOutputs = get_plugin_outputs();
-
+    size_t blobSizeInitial = -1;
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:245]:
```suggestion
    size_t blobCountInitial = -1;
```

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:256]: this outdated logic here has been updated.

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp [hunk 4]
```diff
@@ -242,22 +242,34 @@ void CompileModelCacheTestBase::run() {
...
         {
             core->set_property(ov::cache_dir(m_cacheFolderName));
             ASSERT_NO_THROW(compiledModel = core->compile_model(function, targetDevice, configuration));
-            if (targetDevice.find("AUTO") == std::string::npos)
-                // Apply check only for HW plugins
-                ASSERT_EQ(i != 0, compiledModel.get_property(ov::loaded_from_cache));
+            ASSERT_EQ(i != 0, compiledModel.get_property(ov::loaded_from_cache));
+            while (targetDevice.find("AUTO") != std::string::npos) {
+                auto exeDevices = compiledModel.get_property(ov::execution_devices);
+                if (exeDevices.size() == 1 && exeDevices.front().find("(CPU)") != std::string::npos)
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:257]: Please, remove hardcode -> `exeDevices.front().find("(CPU)")`

Comment (songbell) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:257]: @yangwang201911 seems some problem here, if GPU is compiling faster than cpu help, the logic below will fail.
let's put it this way, for AUTO, first iter : we just check the cache num >0,  at the very end , destroy the compile model to let auto finish async compiling, then record the cache num size
 second iter: compile and compare the result, destroy the compile model, then compare the cache num to the previously recorded one

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp]: Updated.

Comment (yangwang201911) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp]: @songbell Updated.

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:254]: keep this logic for those plugins that didn't support `ov::loaded_from_cache` yet.

**Review by songbell — APPROVED** [review]

**Pull request merged**